packages/queued-request-controller/src/QueuedRequestController.test.ts

```diff
@@ -33,176 +34,654 @@ describe('QueuedRequestController', () => {
       const firstRequest = controller.enqueueRequest(
         buildRequest(),
-        () => firstRequestProcessing,
-      );
-      const secondRequest = controller.enqueueRequest(
-        buildRequest(),
-        async () => {
-          expect(controller.state.queuedRequestCount).toBe(1);
-        },
+        () => new Promise((resolve) => setTimeout(resolve, 10)),
       );
+      // ensure first request skips queue
+      expect(controller.state.queuedRequestCount).toBe(0);
 
-      expect(controller.state.queuedRequestCount).toBe(2);
+      await controller.enqueueRequest(buildRequest(), async () => {
+        expect(controller.state.queuedRequestCount).toBe(0);
+      });
+      expect(controller.state.queuedRequestCount).toBe(0);
 
-      resolveFirstRequest();
       await firstRequest;
-      await secondRequest;
-    });
-
-    it('runs the next request immediately when the queue is empty', async () => {
-      const options: QueuedRequestControllerOptions = {
-        messenger: buildQueuedRequestControllerMessenger(),
-      };
-
-      const controller = new QueuedRequestController(options);
-
-      // Mock requestNext function
-      const requestNext = jest.fn(() => Promise.resolve());
-
-      await controller.enqueueRequest(buildRequest(), requestNext);
-
-      // Expect that the request was called
-      expect(requestNext).toHaveBeenCalledTimes(1);
     });
 
     it('switches network if a request comes in for a different selected chain', async () => {
       const mockSetActiveNetwork = jest.fn();
+      const mockAddRequest = jest.fn();
       const { messenger } = buildControllerMessenger({
+        approvalControllerAddRequest:
+          mockAddRequest.mockResolvedValue(undefined),
         networkControllerGetState: jest.fn().mockReturnValue({
           ...cloneDeep(defaultNetworkState),
           selectedNetworkClientId: 'selectedNetworkClientId',
         }),
         networkControllerSetActiveNetwork: mockSetActiveNetwork,
+        selectedNetworkControllerGetNetworkClientIdForDomain: jest
+          .fn()
+          .mockImplementation((_origin) => 'differentNetworkClientId'),
       });
       const options: QueuedRequestControllerOptions = {
         messenger: buildQueuedRequestControllerMessenger(messenger),
       };
       const controller = new QueuedRequestController(options);
 
       await controller.enqueueRequest(
-        {
-          ...buildRequest(),
-          networkClientId: 'differentNetworkClientId',
-        },
+        buildRequest(),
         () => new Promise((resolve) => setTimeout(resolve, 10)),
       );
 
       expect(mockSetActiveNetwork).toHaveBeenCalledWith(
         'differentNetworkClientId',
       );
+      expect(mockAddRequest).toHaveBeenCalled();
     });
 
-    it('does not switch networks if a request comes in for the same chain', async () => {
+    it('does not switch networks if a request comes in for the same selected chain', async () => {
```

Comment:
these scenarios really should be about same network client not chainId. non important nit

Reply:
Good point! Addressed here: 769b1be3